File: packages/core/src/domain/system_info.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct SystemInfo {
    pub processor: ProcessorInfo,
    pub memory: MemoryInfo,
    pub storage: StorageInfo,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ProcessorInfo {
    pub name: String,
    pub cores: u32,
    pub frequency_ghz: f64,
    pub architecture: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct MemoryInfo {
    pub total_bytes: u64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct StorageInfo {
    pub disks: Vec<DiskInfo>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct DiskInfo {
    pub name: String,
    pub mount_point: String,
    pub total_bytes: u64,
    pub available_bytes: u64,
    pub disk_type: String,
}

File: packages/gui/src/lib/types/system_info.ts
export interface SystemInfo {
  processor: ProcessorInfo;
  memory: MemoryInfo;
  storage: StorageInfo;
}

export interface ProcessorInfo {
  name: string;
  cores: number;
  frequency_ghz: number;
  architecture: string;
}

export interface MemoryInfo {
  total_bytes: number;
  total_display: string;
}

export interface StorageInfo {
  disks: DiskInfo[];
}

export interface DiskInfo {
  name: string;
  mount_point: string;
  total_bytes: number;
  available_bytes: number;
  total_display: string;
  available_display: string;
  disk_type: string;
}
